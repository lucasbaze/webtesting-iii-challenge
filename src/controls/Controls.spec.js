// Test away!
import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import { toBeInTheDocument, toBeDisabled } from '@testing-library/jest-dom';

import Controls from './Controls';

expect.extend({ toBeInTheDocument, toBeDisabled });
afterEach(cleanup);

test('provide buttons to toggle the closed and locked states.', () => {
    expect(shallow(<Controls />)).toMatchSnapshot();
});

test('the closed toggle button is disabled if the gate is locked', () => {
    const { queryByTestId } = render(<Controls locked={true} />);
    expect(queryByTestId('open-close')).toBeDisabled();
});

test('the locked toggle button is disabled if the gate is open', () => {
    const { queryByTestId } = render(<Controls closed={false} />);
    expect(queryByTestId('lock-unlock')).toBeDisabled();
});
