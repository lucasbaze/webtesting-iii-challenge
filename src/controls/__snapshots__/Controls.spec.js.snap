// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provide buttons to toggle the closed and locked states. 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lock-unlock"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="open-close"
  >
    Close Gate
  </button>
</div>
`;
